docs/WALKTHROUGH.md

```diff
@@ -0,0 +1,408 @@
+# Service Catalog Demonstration Walkthrough
+
+This document outlines the basic features of the service catalog by walking
+through a basic use case.
+
+## Step 0 - Prerequisites
+
+### Starting Kubernetes with DNS
+
+You *must* have a Kubernetes cluster with cluster DNS enabled.
+
+* If you are using Google Container Engine or minikube, no additional action is
+required.
+* If you are using hack/local-up-cluster.sh, ensure to set
+KUBE_ENABLE_CLUSTER_DNS:
+
+```console
+KUBE_ENABLE_CLUSTER_DNS=true hack/local-up-cluster.sh -O
+```
+
+### Getting Helm and installing Tiller
+
+If you already have Helm v2 or newer, executing `helm init` should be all that's
+required. Detailed helm installation instructions are available
+[here](https://github.com/kubernetes/helm/blob/master/docs/install.md).
+
+## Step 1 - Installing the Service Catalog System
+
+The service catalog is conveniently packaged as a [helm](http://helm.sh/) 
+chart for installation.
+
+The chart is located in the [charts/catalog](../charts/catalog) directory in this
+repository, and supports a wide variety of customizations which are laid out in
+the README.md in that directory. To install the service-catalog with sensible
+defaults, execute this command from within the Kubernetes directory:
+
+```console
+helm install <path to service-catalog>/charts/catalog --name catalog --namespace catalog
+```
+
+Note: in the event you need to start the walk through over, the easiest way is
+to execute `helm delete --purge catalog`.
+
+## Step 2 - Understand Service Catalog Components
+
+Now that the system has been deployed to our Kubernetes cluster, we can use
+`kubectl` to talk to the service catalog API server.  The service catalog API
+has four resources:
+
+- Broker Server: A server that acts as a service broker and conforms to the 
+    [Open Service Broker API](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md)
+- `Broker` Resource: a representation of the broker server in the service-catalog system
+    that indicates where a broker server is running
+- `ServiceClass`: a service offered by a particular service broker. `ServiceClass`es are created
+    in response to a `Broker` resource being submitted to the system
+- `Instance`: an instance of a `ServiceClass` provisioned by the `Broker` for that `ServiceClass`
+- `Binding`: a binding to an `Instance` which is manifested into a Kubernetes namespace
+
+These components are building blocks of the service catalog in Kubernetes from an
+API standpoint.
+
+----
+
+#### Note: accessing the service catalog
+
+Unfortunately, `kubectl` doesn't know how to speak to both the service catalog
+API server and the main Kubernetes API server without switching contexts or
+`kubeconfig` files.  For now, the best way to access the service catalog API
+server is via a dedicated `kubeconfig` file.  You can manage the kubeconfig in
+use within a directory using the `direnv` tool.
```

Comment:
I know we have some `direnv` fans of on the team, but I am wondering if this truly is the best option here. The practical effect of saying so is that it's going to send readers off on a tangent, installing and setting up another tool.

Can I suggest that after mentioning the need for some kubectl context switching, we flow right into the next section (creating kubecontext) and then for all relevant kubectl commands throughout the remainder of the walkthrough, we include the `--context service-catalog` flag?

I feel this will be far more comprehensive.